### PR TITLE
AzureMonitor: Update docs for current user authentication

### LINKED ISSUE #85640: AzureMonitor: Document user-authentication
See PR for more details

### PR DESCRIPTION
**Pull request by aangelisc**
Updates documentation to provide additional detail on current user authentication in Azure Monitor.

Part of https://github.com/grafana/grafana/pull/81918

Fixes #85640

### CODE REVIEW

**Review by taleena — COMMENTED** [review]
Sent a few suggestions questions via Slack, otherwise looks good, can approve after reading through that. 👍

**Review by taleena — APPROVED** [review]
Changes reviewed, additions look good, thank you!

**Review by adamyeats — APPROVED** [review]

File: docs/sources/datasources/azure-monitor/_index.md [hunk 1]
```diff
@@ -133,6 +133,25 @@ datasources:
     version: 1
 ```
 
+**Current User:**
+
+**Note: The `oauthPassThru` property is required for current user authentication to function. Additionally, `disableGrafanaCache` is necessary to ensure cached responses to resources users do not have access to in Azure are not returned.`**
```
Comment (lwandz13) [inline docs/sources/datasources/azure-monitor/_index.md:138]:
```suggestion
{{< admonition type="note" >}}
The `oauthPassThru` property is required for current user authentication to function. Additionally, `disableGrafanaCache` is necessary to ensure cached responses to resources users do not have access to in Azure are not returned.
{{< /admonition >}}
```

File: docs/sources/datasources/azure-monitor/_index.md [hunk 2]
```diff
@@ -141,6 +160,8 @@ datasources:
 | **Microsoft Chinese national cloud** | `chinaazuremonitor`        |
 | **US Government cloud**              | `govazuremonitor`          |
 
+**Note: Cloud names for current user authentication differ from the above. The public cloud name is `AzureCloud`, the Chinese national cloud name is `AzureChinaCloud`, and the US Government cloud name is `AzureUSGovernment`.**
```
Comment (lwandz13) [inline docs/sources/datasources/azure-monitor/_index.md:163]:
```suggestion
{{< admonition type="note" >}}
Cloud names for current user authentication differ from the above. The public cloud name is `AzureCloud`, the Chinese national cloud name is `AzureChinaCloud`, and the US Government cloud name is `AzureUSGovernment`.
{{< /admonition >}}
```

File: docs/sources/datasources/azure-monitor/_index.md [hunk 3]
```diff
@@ -201,6 +222,35 @@ For details on workload identity, refer to the [Azure workload identity document
    workload_identity_token_file = TOKEN_FILE_PATH
    ```
 
+### Configure Current User Authentication
+
+**Note: This feature is experimental and subject to change or removal. Aspects of Grafana may not work as expected when using this authentication method.**
```
Comment (lwandz13) [inline docs/sources/datasources/azure-monitor/_index.md:227]:
```suggestion
{{< admonition type="note" >}}
This feature is experimental and subject to change or removal. Aspects of Grafana may not work as expected when using this authentication method.
{{< /admonition >}}
```

File: docs/sources/datasources/azure-monitor/_index.md [hunk 4]
```diff
@@ -201,6 +222,35 @@ For details on workload identity, refer to the [Azure workload identity document
...
+
+This method of authentication does not inherently support backend functionality as a user's credentials will not be in scope e.g. alerting, reporting, recorded queries. In order to support backend queries when using a data source configured with current user authentication, it is possible to configure service credentials. Also, note that query and resource caching will be disabled by default for data sources using current user authentication.
+
+**Note: To configure fallback service credentials the [feature toggle][configure-grafana-feature-toggles] `idForwarding` must be set to `true` and `user_identity_fallback_credentials_enabled` must be enabled in the [Azure configuration section][configure-grafana-azure] (enabled by default when `user_identity_enabled` is set to `true`).**
```
Comment (lwandz13) [inline docs/sources/datasources/azure-monitor/_index.md:233]:
```suggestion
{{< admonition type="note" >}}
To configure fallback service credentials the [feature toggle][configure-grafana-feature-toggles] `idForwarding` must be set to `true` and `user_identity_fallback_credentials_enabled` must be enabled in the [Azure configuration section][configure-grafana-azure] (enabled by default when `user_identity_enabled` is set to `true`).
{{< /admonition >}}
```

**Review by lwandz13 — DISMISSED** [review]

**Pull request merged**